students/RimleeT/lesson05/mailroom3.py

```diff
@@ -0,0 +1,125 @@
+import sys
+import os
+import tempfile
+donor_db = {"William Gates, III": [653772.32, 12.17],   
+            "Jeff Bezos": [877.33],
+            "Paul Allen": [663.23, 43.87, 1.32],
+            "Mark Zuckerberg": [1663.23, 4300.87, 10432.0],
+}
+
+"""
+Send a Thank You
+If the user (you) selects “Send a Thank You” option, prompt for a Full Name.
+If the user types 'list' show them a list of the donor names and re-prompt.
+If the user types a name not in the list, add that name to the data structure and use it.
+If the user types a name in the list, use it.
+Once a name has been selected, prompt for a donation amount.
+Convert the amount into a number; it is OK at this point for the program to crash if someone types a bogus amount.
+Add that amount to the donation history of the selected user.
+Finally, use string formatting to compose an email thanking the donor for their generous donation. Print the email to the terminal and return to the original prompt.
+It is fine (for now) for the program not to store the names of the new donors that had been added, in other words, to forget new donors once the script quits running.
+"""	
+
+def do_thankyou():
+    print("<<<SEND A THANK YOU>>>")
+                
+    #if user types 'list' show the list of donors and prompt to enter donor name
+    donor_name = None
+    while True:
+        donor_name = input("Enter the name of donor or type 'list' if you wish to view the list of donors: ")
+        if donor_name.lower() == 'list':
+            print(donor_db)
+        else:
+            break
+    donation_amount = get_donation_amount() 
+
+    if donor_name not in donor_db:
+        donor_db[donor_name] = [donation_amount]
+    else:
+        donor_db[donor_name].append(donation_amount)
+    
+    print("donor_db",donor_db)
+    print (f"Thank you {donor_name} for your generous donation amount of ${donation_amount}" )
+
+def get_donation_amount():
+    try:
+        return float(input("Enter the donation amount: "))
+    except:
```

Comment:
This will catch your error but we usually want to avoid general except catchers. We want to catch specific ones so that we are aware of the specific error we see. Otherwise you could be catching errors that are not expected and run into others issues